libtiledbsoma/test/test_indexer.py

```diff
@@ -0,0 +1,193 @@
+import numpy as np
+import pandas as pd
+import tiledb
+from tiledbsoma.utils import build_index
+from tiledbsoma.options import SOMATileDBContext
+from tiledbsoma.options._soma_tiledb_context import _validate_soma_tiledb_context
+
+
+# tiledbsoma.pytiledbsoma.config_logging("debug")
+
+
+# 1d array to list
+
+
+def indexer_test(keys: np.array, lookups: np.array, fail: bool):
+    if fail:
+        indexer_test_fail(keys, lookups)
+    else:
+        indexer_test_pass(keys, lookups)
+
+
+def indexer_test_fail(keys: np.array, lookups: np.array):
+    try:
+        context = _validate_soma_tiledb_context(SOMATileDBContext(tiledb.default_ctx()))
+        index = build_index(keys, context)
+        index.get_indexer(lookups)
+        raise AssertionError("should have failed")
+    except pd.errors.InvalidIndexError:
+        pass
+
+    try:
+        pd_index = pd.Index(keys)
+        pd_index.get_indexer(lookups)
+        raise AssertionError("should have failed")
+    except pd.errors.InvalidIndexError:
+        pass
```

Comment:
https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest.raises

There is a way to test for expected exceptions in Pytest by using `with pytest.raises(...):`

Reply:
Are these tests also run with pytest? I hadn’t realized that; I thought they were just executed as `python the-test-file.py`.

Reply:
Yeah when you [run](https://github.com/single-cell-data/TileDB-SOMA/blob/main/Makefile#L35) `make test` it also runs all the Python unit tests in `libtiledbsoma/test`.

Reply:
Oh wait I'm looking at this file again, and I see what you're saying now. Why isn't this written for Pytest?